Tactics/Common.lean

```diff
@@ -98,18 +98,30 @@ def getBitVecValue? (e : Expr) : MetaM (Option ((n : Nat) × BitVec n)) :=
     | _ => Lean.Meta.getBitVecValue? e
 
 /-- Given a ground term `e` of type `Nat`, fully reduce it,
-and attempt to reflect it into a meta-level `Nat` -/
+and attempt to reflect it into a meta-level `Nat`,
+returning `none` on failure -/
+partial def reflectNatLiteral? (e : Expr) : MetaM (Option Nat) := OptionT.run do
+  let e' ← instantiateMVars e
```

Comment:
I guess I consider an mvar that has been assigned (but not yet instantiated) to be morally ground.
I *think* `instantiateMVars` has been optimized so that it's basically free on an expression that does not have mvars, so there shouldn't be much of an overhead to having it